### PR TITLE
MAINT: Add copyright to the top of each file

### PR DESCRIPTION
**Pull request by larsoner**
This script `tools/dev/ensure_headers.py` reads all `.py` files in our source tree and ensures that our standard license line shows up followed by a new copyright line suggested by [bestpractices.dev](https://www.bestpractices.dev/en/projects/7783?criteria_level=2). Only takes a few hundred milliseconds to run, and running it more than once is a no-op. Eventually we could maybe turn it into a pre-commit hook but for now we can probably just add it to our release instructions as yet another step. It provides a framework for eventually (in another PR) tackling #11605.

### CODE REVIEW

**Comment by larsoner** [comment]
Leaving this as draft mode just to avoid pinging pretty much the entire steering council :) @drammock feel free to take a look and see if it makes sense to you

File: tools/dev/ensure_headers.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    for entry in root:
+        if entry.type == "tree":
+            for x in get_paths_from_tree(entry, level + 1):
+                yield x
+        else:
+            yield Path(entry.path)  # entry.type
+
+
+def _ensure_license(lines, path):
+    # 1. Keep existing
```
Comment (larsoner) [inline tools/dev/ensure_headers.py:42]: @drammock this is the one new file. Might be easiest to check out my branch and look at `tools/dev/ensure_headers.py` locally. GitHub is really slow with a diff across so many files

**Comment by drammock** [comment]
should we add `gitpython` as a new dev dependency in pyproject.toml? I'm OK with that but also OK omitting it since it's only the release manager du jour that needs it. I looked at the script on your fork as suggested, LGTM.

**Review by drammock — APPROVED** [review]

**Review by mscheltienne — APPROVED** [review]

**Comment by cbrnr** [comment]
Shouldn't it be "Copyright **by** MNE-Python contributors."?

**Review by sappelhoff — APPROVED** [review]

**Comment by sappelhoff** [comment]
> Shouldn't it be "Copyright by the MNE-Python contributors."?

the wording here is the one suggested here: https://www.bestpractices.dev/en/projects/7783?criteria_level=2

I'd be +1 to keep in line with their "template"

**Comment by larsoner** [comment]
Pushed to `main` as https://github.com/mne-tools/mne-python/commit/5897b818e90308d25713fd69552fd29862757dce

**Pull request closed without merging**